### PR TITLE
[ARM] `az group delete`: Add new parameter `--force-deletion-types` to support force deletion

### PR DESCRIPTION
**Pull request by tjegbejimba**
**Description**

This PR enables the force deletion types parameter to group deletion. Updated az group delete test cases
**Testing Guide**

az group delete -n testRG --force-deletion-types Microsoft.Compute/virtualMachines

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 1]
```diff
@@ -504,6 +504,7 @@ def load_arguments(self, _):
     with self.argument_context('group delete') as c:
         c.argument('resource_group_name', resource_group_name_type,
                    options_list=['--name', '-n', '--resource-group', '-g'], local_context_attribute=None)
+        c.argument('force_deletion_types', options_list=['--forceDeletionTypes', '-f'], min_api='2021-04-01', help='The resource types you want to force delete. Currently, only the following is supported: forceDeletionTypes=Microsoft.Compute/virtualMachines,Microsoft.Compute/virtualMachineScaleSets')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:507]:
```suggestion
        c.argument('force_deletion_types', options_list=['--force-deletion-types', '-f'], min_api='2021-04-01', help='The resource types you want to force delete. Currently, only the following is supported: forceDeletionTypes=Microsoft.Compute/virtualMachines,Microsoft.Compute/virtualMachineScaleSets')
```

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 2]
```diff
@@ -504,6 +504,7 @@ def load_arguments(self, _):
     with self.argument_context('group delete') as c:
         c.argument('resource_group_name', resource_group_name_type,
                    options_list=['--name', '-n', '--resource-group', '-g'], local_context_attribute=None)
+        c.argument('force_deletion_types', options_list=['--force-deletion-types', '-f'], min_api='2021-04-01', help='The resource types you want to force delete. Currently, only the following is supported: forceDeletionTypes=Microsoft.Compute/virtualMachines,Microsoft.Compute/virtualMachineScaleSets')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:507]: Could you use `arg_type=get_enum_type(xxx)` to define enumerations for these valid values?

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 3]
```diff
@@ -504,6 +504,7 @@ def load_arguments(self, _):
     with self.argument_context('group delete') as c:
         c.argument('resource_group_name', resource_group_name_type,
                    options_list=['--name', '-n', '--resource-group', '-g'], local_context_attribute=None)
+        c.argument('force_deletion_types', options_list=['--force-deletion-types', '-f'], arg_type=get_enum_type("Microsoft.Compute/virtualMachines", "Microsoft.Compute/virtualMachineScaleSets"), min_api='2021-04-01', help='The resource types you want to force delete. Currently, only the following is supported: forceDeletionTypes=Microsoft.Compute/virtualMachines,Microsoft.Compute/virtualMachineScaleSets')
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:507]:
```suggestion
        c.argument('force_deletion_types', options_list=['--force-deletion-types', '-f'], arg_type=get_enum_type(['Microsoft.Compute/virtualMachines', 'Microsoft.Compute/virtualMachineScaleSets']), min_api='2021-04-01', help='The resource types you want to force delete.')
```

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**